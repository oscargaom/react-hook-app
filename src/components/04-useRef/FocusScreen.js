import React, { useRef } from 'react'
import "../../styles/effects.css";

export const FocusScreen = () => {
    
    const refCtrl = useRef();
    console.log(refCtrl);

    const handleClick = () => {
        // Pone el foco en el input
        // document.querySelector("input").focus();
        // Pone el foco en el input y adem√°s selecciona el contenido del mismo
        // document.querySelector("input").focus();
        // Esta es otra manera de hacerlo usando el useRef
        refCtrl.current.select();
    };

    return (
        <div>
            <h1>Focus Screen</h1>
            <hr/>

            <input
                type="text"
                ref={refCtrl}
                className="form-control"
                placeholder="Introduce tu nombre"
            />

            <button
                onClick={handleClick} 
                className="btn btn-outline-primary mt-3">
                Focus
            </button>
        </div>
    )
}
