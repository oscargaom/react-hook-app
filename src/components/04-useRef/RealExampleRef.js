import React, { useState } from 'react'
import "../../styles/effects.css"
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks'

export const RealExampleRef = () => {

    const [show, setShow] = useState(false);

    return (
        <div>
            <h1>Real Example Ref</h1>            
            <hr/>

            {show && <MultipleCustomHooks/>}

            <button
                className="btn btn-outline-primary mt-3"
                onClick={() => setShow(!show) }
            >
                Show/Hide
            </button>
        </div>
    )
}
