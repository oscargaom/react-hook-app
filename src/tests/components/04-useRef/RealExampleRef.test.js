import React, { useState } from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';


describe('Pruebas del componente RealExampleRef', () => {
    
    test('Debe mostrarse correctamente', () => {
        const wrapper = shallow(<RealExampleRef />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
    });

    

    test('Debe de mostrar el componente MultipleCustomHooks (show = true)', () => {
        
        const wrapper = shallow( <RealExampleRef/>);

        // console.log(wrapper.html());
         wrapper.find('button').simulate('click');
        // console.log(wrapper.html());

        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    });
})
