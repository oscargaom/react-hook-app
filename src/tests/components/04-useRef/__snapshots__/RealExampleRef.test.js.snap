// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente RealExampleRef Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
